tests/pgautofailover_utils.py

```diff
@@ -93,19 +94,18 @@ def pg_createcluster(self, datadir, port=5432):
         directory path.
         """
         vnode = self.vlan.create_node()
-        nodeid = len(self.datanodes) + 1
 
         create_command = ["sudo", shutil.which('pg_createcluster'),
                           "-U", os.getenv("USER"),
                           PGVERSION, datadir, '-p', str(port)]
 
         print("%s" % " ".join(create_command))
 
-        create_proc = vnode.run(create_command)
-        out, err = create_proc.communicate(timeout=COMMAND_TIMEOUT)
-        if create_proc.returncode > 0:
-            raise Exception("pg_createcluster failed, out: %s\n, err: %s" %
-                            (out, err))
+        with vnode.run(create_command) as create_proc:
+            out, err = create_proc.communicate(timeout=COMMAND_TIMEOUT)
+            if create_proc.returncode > 0:
+                raise Exception("pg_createcluster failed, out: %s\n, err: %s" %
+                                (out, err))
```

Comment:
Could this be `self.vnode.run_and_wait(create_command, name="pg_createcluster")`?